src/appengine/handlers/cron/fuzz_strategy_selection.py

```diff
@@ -0,0 +1,135 @@
+metric to be for edges, crash, features, or units. Currently based on new
+edges."""
+
+from datastore import data_types
+from datastore import ndb
+from datastore import ndb_utils
+from google_cloud_utils import big_query
+from handlers import base_handler
+from libs import handler
+
+# BigQuery query for calculating multi-armed bandit probabilities for
+# various strategies using a Boltzman Exploration (softmax) model.
+
+# Averages standardized new_edges feature over each strategy for expected
+# new_edges metric for each strategy.
+# See https://www.cs.mcgill.ca/~vkules/bandits.pdf for formula.
+
+BANDIT_PROBABILITY_QUERY = """
+SELECT
+  /* Calculate bandit weights from calculated exponential values. */
+  strategy,
+  strategy_exp / exp_sum AS bandit_weight,
+  strategy_count
+FROM
+  (SELECT
+    EXP(strategy_avg_edges / temperature) AS strategy_exp,
+    SUM(EXP(strategy_avg_edges / temperature)) OVER() AS exp_sum,
+    strategy,
+    strategy_count
+  FROM
+    (SELECT
+      /* Standardize the new edges data and take averages per strategy. */
+      AVG((new_edges - overall_avg_new_edges) / overall_stddev_new_edges) AS strategy_avg_edges,
+      strategy,
+      /* Change temperature parameter here. */
+      .5 as temperature,
+      COUNT(*) AS strategy_count
+    FROM
+      (SELECT
+        fuzzer,
+        CONCAT(s_radamsa, s_max_len, s_ml_rnn, s_vp, s_corpus, s_fork, s_subset, s_recommended_dict) AS strategy,
+        fuzzer_stddev,
+        AVG(new_edges) OVER() AS overall_avg_new_edges,
+        STDDEV(new_edges) OVER() AS overall_stddev_new_edges,
+        new_edges
+      FROM
+        (SELECT
+          fuzzer,
+          IF(strategy_corpus_mutations_radamsa > 0, "radamsa,", "") AS s_radamsa,
+          IF(strategy_random_max_len > 0, "max len,", "") AS s_max_len,
+          IF(strategy_corpus_mutations_ml_rnn > 0,"ml rnn,", "") AS s_ml_rnn, 
+          IF(strategy_value_profile > 0, "value profile,", "") AS s_vp, 
+          IF(strategy_corpus_mutations > 0, "corpus,", "") AS s_corpus,
+          IF(strategy_fork > 0, "fork,", "") AS s_fork,
+          IF(strategy_corpus_subset > 0, "subset,", "") AS s_subset,
+          IF(strategy_recommended_dict > 0, "dict,", "") AS s_recommended_dict,
+          STDDEV(new_edges) OVER(PARTITION by fuzzer) AS fuzzer_stddev,
+          new_edges
+        FROM 
+          libFuzzer_stats.TestcaseRun
+        WHERE
+           ((strategy_mutator_plugin = 0) OR (strategy_mutator_plugin IS NULL)) AND
+            DATE_DIFF(cast(current_timestamp() AS DATE), cast(_PARTITIONTIME AS DATE), DAY) < 31 AND
+            ((strategy_corpus_mutations = 0) OR (strategy_corpus_mutations IS NULL)) AND
+            ((strategy_handle_unstable = 0) OR (strategy_handle_unstable IS NULL)) AND
+            ((strategy_weighted_mutations = 0) OR (strategy_weighted_mutations IS NULL)))
+      WHERE
+        /* Filter for unstable targets. */
+        fuzzer_stddev < 150)
```

Comment:
nit: is this meant to be `150`? I recall Abhishek mentioning `50`, but I wasn't following up on that, just want to clarify.

Reply:
We looked through the data beforehand and 150 seemed to do a pretty good job of including only targets we knew would be problematic.

Reply:
Got it, awesome!